webapp/lib/deploymentProvider/PublishProfileDeploymentProvider.js

```diff
@@ -93,6 +93,9 @@ class PublishProfileDeploymentProvider {
                 username: res.userName,
                 password: res.userPWD
             };
+            console.log(`::add-mask::${creds.username}`);
+            console.log(`::add-mask::${creds.password}`);
+            console.log(`::add-mask::${creds.uri}`);

```

Comment:
Why's this even required? if this just going to print some extra `***`s in your logs?

Reply:
Are you asking about the masking of `creds.uri` or all 3 collectively?

I'm not sure all of what needs to be masked, I did all 3 to be safe. The overall change is needed so that we don't print credentials though. This will replace any instances of the strings (username/password/uri) that appear in this step or elsewhere, so it ensures that they don't get leaked

Reply:
All 3 collectively. 

Okay, if I get this right, if any action takes a complex object as an input, the action-author will have to add these steps to explicitly to call out the secrets. To ensure masking of those occurrences elsewhere?

Reply:
Correct. We don't mask any secret substrings or try to parse secrets into an object. We may change some of that behavior, but the safest thing is always to explicitly mask anything that needs to be masked

Reply:
Sounds good. Is there any documentation for this?

Reply:
https://github.com/actions/toolkit/blob/master/docs/commands.md#masking-values-in-logs

There's not docs explicitly calling out this behavior, AFAIK, we're working on making those better/improving the model

Reply:
Here are the user facing docs:
https://help.github.com/en/articles/virtual-environments-for-github-actions#creating-and-using-secrets-encrypted-variables

From those docs:

> Secret names cannot include any spaces. To ensure that GitHub redacts your secret in logs, avoid using structured data as the values of secrets, like JSON or encoded Git blobs.